decoder.cpp

```diff
@@ -190,6 +198,108 @@ bool Decoder::readImageResources()
   return (length == 0);
 }
 
+double Decoder::getTypeNumber(const std::unique_ptr<OSType>& v)
```

Comment:
It looks like a candidate for a virtual member function in OSType:
```
struct OSType {
  virtual double numberValue() { return 0.0; }
}
```
And override it on OSTypeLargeInt, OSTypeInt, etc.